### ISSUE TITLE
Best Div "time ago" not updating

### ISSUE THREAD

**Issue by ezio416** [openplanet-nl/tmio#72]
In COTD reruns, I've gotten div 2 before, but today I got it again and improved my best placement in it from 7th to 2nd. My new best placement was updated, but the time it says I got my best div still says it was 5 months ago, implying that I got 2nd place at that time.

https://trackmania.io/#/player/ezio/cotd
<img width="290" alt="Screenshot 2023-10-26 231403" src="https://github.com/openplanet-nl/tmio/assets/37951898/409250fd-ef4a-4144-b693-c8208fc7ace1">

**Comment by ezio416** [comment]
For reference, in the main cup, I had won div 22 a while ago, then at a later date improved my best to win div 20 and the time listed under "best div rank" updated to reflect that (even though it was still 1st), so I think all these "time ago" texts should be consistent with any improvement.

**Comment by davidbmaier** [comment]
Good catch, there's a mismatch between the best division stat and the rank that gets displayed there. Should be a simple fix.

Regarding whether these stats should update if the same is achieved again later, I would say the most intuitive way for these stats to work is to show the first instance that rank/div was achieved, no matter if there is a more recent instance or one in a higher div (in the case of div ranks).
Otherwise people that routinely get division 1 would look like they only yesterday got their best division ever (when they've possibly been in div 1 consistently for a year or so). If you want to see recent performance, you can just scroll down.

Right now that's how it works for *Best rank* and *Best div*, but *Best div rank* specifically shows the best division rank **in the highest div** (so if you got 1st in div 22 and later get 1st in div 20, it'll update to that div 20 win).
In my opinion that feels inconsistent when compared to the other stats, and I'd prefer to just show the first time that best div rank was achieved, no matter the div.

This logic was originally implemented by @codecat, maybe she can add some context to that decision (and why the other stats don't update when the same is achieved again).
My suggestion would be to handle the *Best div rank* stat the same as the others and only show the very first time that exact div rank was achieved - but I'm open for other opinions, in the end the point of the page is to show data that's useful and interesting to the players.
There's also an argument for just making both variants available and letting the user choose, but I'm not sure it's worth the effort (given that nobody's really asked for that kind of change as far as I'm aware).

Whatever the decision is, I'll probably add a clarification note to the tooltips either way so we don't need to guess what the stats really mean.

**Comment by codecat** [comment]
> Best div rank specifically shows the best division rank in the highest div

The idea was that it would show the highest rank in any division, not the best division. For example, at the time of writing, [my profile](https://trackmania.io/#/player/miss/cotd) says my best division was 2 (placed 11th 3 years ago), but my best div rank was 6th (in div 11, 2 years ago).

The logic is a bit confusing I must admit. I'm open to changing it.

**Comment by davidbmaier** [comment]
Just to clarify, I meant *Best rank in div* shows the best rank in the best div you got it in (not just the best div overall of course, that's what *Best div* is for). So if you had gotten 6th in a lower div, it would show that instead of the first time you got 6th.

**Comment by davidbmaier** [comment]
These changes have been merged and deployed, your reported issues should be fixed (along with some extra tooltips that should better explain the *Best div rank* stat).

Feel free to take a look and confirm it yourself @ezio416 in case I missed something.

**Comment by ezio416** [comment]
@davidbmaier A few things seem broken. Now it's showing that my placement of 18th in div 2 is my best (which is from the first time I got div 2 in reruns - far from my best), and in the rerun panel it's showing I got 1st of div 22 (when I've won div 4, and div 22 obviously isn't a rerun). It's also showing the div 22 win under main cups as being my best when I've won div 20 (22 was my first win). It seems like it's now just taking the very first result that matches the div/div rank. Is this expected?

edit - yeah based on the new tooltips this is expected, but I'm kinda sad it doesn't actually show my best results anymore (i.e. my rerun div 4 and main div 20 wins no longer showing)

**Comment by davidbmaier** [comment]
I'd say the current logic is more consistent, but I get your point.
To be honest I feel like the whole stats might deserve a general overhaul to be more dynamic.

The "Best div rank" stat isn't really meant to show the best overall rank (or highest div you got the best div rank in), it's just supposed to show what your highest rank within a div was (which obviously becomes a bit pointless once you have won a div).
So similarly to how we hide all the stats for people that have won div 1, it might make more sense to adapt that whole statistic when the player has won a division (e.g. by replacing it with a "Best div win" stat).

Additionally I'm thinking a switch to change between "first instance" and "most recent instance" could be cool for these stats (since duplicates can clearly occur). But that probably needs some more thinking, especially if we do actually hide some of that data.

I'll reopen this issue for now (since we already have some discussion in here) but make it an enhancement item instead of a bug - and I'll hopefully find some time to look into this soon.
Feel free to add ideas if you have some, maybe we can think of extra stats that could be shown.

**Comment by davidbmaier** [comment]
I've added both the latest and the first instance to the stats, maybe that's a good first compromise. Would be interested to hear what you guys think and if this solves the main issues, or if there's still some missing data you'd like to see in this context.
